src/main/java/org/gridsuite/useradmin/server/repository/AnnouncementRepository.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.useradmin.server.repository;
+
+import org.gridsuite.useradmin.server.entity.AnnouncementEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+@Repository
+public interface AnnouncementRepository extends JpaRepository<AnnouncementEntity, UUID> {
+
+    @Query(nativeQuery = true, value = "SELECT * from announcement where start_date <= :endDate and end_date >= :startDate")
+    List<AnnouncementEntity> findOverlappingAnnouncements(Instant startDate, Instant endDate);
+
+    @Query(nativeQuery = true, value = "SELECT * from announcement where start_date < :now and end_date < :now")
+    List<AnnouncementEntity> findExpiredAnnouncements(Instant now);
```

Comment:
Can be simplified as
```suggestion
    boolean existsOverlappingAnnouncementsByStartDateLessThanEqualAndEndDateGreaterThanEqual(@NonNull Instant startDate, @NonNull Instant endDate);

    @Transactional
    @Modifying
    @Query("delete from AnnouncementEntity a where a.startDate < now() and a.endDate < now()")
    int deleteExpiredAnnouncements();
```

_`org.springframework.lang.NonNull` for the annotations_.